### PR TITLE
HPCC-32003 Automated testing of hyperlinks in Documentation files

### PR DESCRIPTION
**Pull request by Charan-Sharan**
… user documents and GitHub README files using GitHub Actions

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: .github/workflows/test-hyperlinks.yml [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    branches:
+      - "master"
+      - "candidate-*"
+      - "!candidate-9.4.*"
+      - "!candidate-9.2.*"
+      - "!candidate-9.0.*"
+      - "!candidate-8.*"
+      - "!candidate-7.*"
+      - "!candidate-6.*"
+    paths:
```
Comment (AttilaVamos) [inline .github/workflows/test-hyperlinks.yml:14]: Are these paths definitions really needed here if you do the same from line 40.

**Review by Michael-Gardner — CHANGES_REQUESTED** [review]
Have you confirmed this is doing what you expect it to do? I was looking over the successful run on https://github.com/hpcc-systems/HPCC-Platform/actions/runs/9672181979/job/26684061123 and I don't see any output, or any link tests being done. Perhaps its not finding the files to even start the process.  Looking at your bash I would have expected it to be outputing link information.

**Comment by Charan-Sharan** [comment]
@Michael-Gardner Yes, it is working as expected. The workflow is made to be triggered and run only for PRs that contain documentation file changes. Since my PR(this PR) didn't had any documentation files changed or added, It didn't list any files. 
To check the complete repository it needs to be triggered by workflow_dispatch. Here is a workflow run, where we made a force run to check all the files of repo.
https://github.com/hpcc-systems/HPCC-Platform/actions/runs/9644440429/job/26596518707

**Review by Michael-Gardner — APPROVED** [review]
Well done. Looks good.  Perhaps we should point @g-pan to the error logs at https://github.com/hpcc-systems/HPCC-Platform/actions/runs/9644440429/artifacts/1631301365 so he can confirm they are relevant and/or get a fix in asap.

**Review by AttilaVamos — APPROVED** [review]
Seem good to go.

**Pull request merged**